onnx/optimizer/pass_manager.cc

```diff
@@ -0,0 +1,46 @@
+#include "onnx/optimizer/pass_manager.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+PassManager::PassManager() {}
+PassManager::~PassManager() {}
+
+GeneralPassManager::~GeneralPassManager() {
+  this->passes.clear();
+}
+void GeneralPassManager::add(std::shared_ptr<Pass> pass) {
+  this->passes.insert(pass);
+}
+
+std::shared_ptr<PassManagerAnalysis> GeneralPassManager::run(Graph& graph) {
+  for (std::shared_ptr<Pass> pass : this->passes) {
+    auto pass_analysis = pass->runPass(graph);
+  }
+  return std::shared_ptr<PassManagerAnalysis>(new EmptyPassManagerAnalysis());
+}
+
+std::shared_ptr<PassManagerAnalysis> FixedPointPassManager::run(Graph& graph) {
+  bool fixed_point_optimization_done;
+
+  do {
+    fixed_point_optimization_done = false;
+    for (std::shared_ptr<Pass> pass : this->passes) {
+      std::shared_ptr<PostPassAnalysis> analysis = pass->runPass(graph);
+      std::shared_ptr<CountBasedPassAnalysis> count_analysis =
+          std::dynamic_pointer_cast<CountBasedPassAnalysis>(analysis);
+
+      while (nullptr != count_analysis &&
+             count_analysis->fixedPointOptimizationNeeded()) {
+        count_analysis = std::dynamic_pointer_cast<CountBasedPassAnalysis>(
```

Comment:
Can we get ride of `dynamic_pointer_cast`? Some projects would like to disable RTTI.